src/raftstore/store/snap.rs

```diff
@@ -618,6 +620,987 @@ mod v1 {
+                    return Err(box_err!("fail to parse snapshot meta key {:?}, err: {:?}",
+                                        key_bytes,
+                                        e));
+                }
+            };
+            let strs: Vec<&str> = key.split('_').collect();
+            if strs.len() != 2 {
+                return Err(box_err!("invalid snapshot meta key {}", key));
+            }
+            let cf_name = strs[1];
+            let res = snapshot_cfs.iter().find(|&s| *s == cf_name);
+            if res.is_none() {
+                return Err(box_err!("invalid snapshot cf {}", cf_name));
+            }
+            let cf = res.unwrap();
+            match strs[0] {
+                SNAPSHOT_META_PREFIX_SIZE => {
+                    let size = try!(kv.get_value().decode_u64());
+                    cf_sizes.push((cf, size));
+                }
+                SNAPSHOT_META_PREFIX_CHECKSUM => {
+                    let value = try!(kv.get_value().decode_u64());
+                    if value > u32::max_value() as u64 {
+                        return Err(box_err!("invalid snapshot checksum {} for cf {}", value, cf));
+                    }
+                    cf_checksums.push((cf, value as u32));
+                }
+                _ => return Err(box_err!("invalid snapshot meta prefix {}", strs[0])),
+            }
+        }
+        if cf_sizes.len() != snapshot_cfs.len() {
+            return Err(box_err!("invalid number of snapshot meta cf size: {}, expected: {}",
+                                cf_sizes.len(),
+                                snapshot_cfs.len()));
+        }
+        if cf_checksums.len() != snapshot_cfs.len() {
+            return Err(box_err!("invalid number of snapshot meta cf checksum: {}, expected: {}",
+                                cf_checksums.len(),
+                                snapshot_cfs.len()));
+        }
+        let mut cf_size_checksums = Vec::with_capacity(snapshot_cfs.len());
+        for (cf, size) in cf_sizes {
+            let checksum = cf_checksums.iter().find(|c| c.0 == cf).unwrap().1;
+            cf_size_checksums.push((cf, size, checksum));
+        }
+        Ok(cf_size_checksums)
+    }
+
+    #[derive(Default)]
+    struct CfFile {
+        pub cf: CfName,
+        pub path: String,
+        pub size: u64,
+
+        // for building snapshot
+        pub sst_writer: Option<SstFileWriter>,
+        pub kv_count: u64,
+        // for sending/receiving snapshot
+        pub file: Option<File>,
+
+        // for writing snapshot
+        pub tmp_path: String,
+        pub written_size: u64,
+
+        pub checksum: u32,
+    }
+
+    #[derive(Default)]
+    struct MetaFile {
+        pub data: RaftSnapshotData,
+        pub path: String,
+        pub file: Option<File>,
+
+        // for writing snapshot
+        pub tmp_path: String,
+    }
+
+    pub struct Snap {
+        display_path: String,
+        cf_files: Vec<CfFile>,
+        cf_index: usize,
```

Comment:
It's used as the index of `cf_files` which point to the current offset of reading/writing cf file.